frontend/web/src/assets/lib/jsc3d/jsc3d.webgl.js

```diff
@@ -0,0 +1,1202 @@
+					'uniform sampler2D s_palette; \n' + 
+					'uniform sampler2D s_texture; \n' + 
+					'uniform sampler2D s_sphereTexture; \n' + 
+					'varying vec3 v_normal; \n' + 
+					'varying vec2 v_texCoord; \n' + 
+					'\n' + 
+					'void main(void) { \n' + 
+					'	vec4 materialColor = u_isLit ? vec4(texture2D(s_palette, vec2(abs(v_normal.z), 0.0)).rgb, u_opacity) : vec4(1.0, 1.0, 1.0, u_opacity); \n' + 
+					'	if(u_isCast) { \n' + 
+					'		gl_FragColor = materialColor * texture2D(s_sphereTexture, vec2(0.5, -0.5) * v_normal.xy + vec2(0.5, 0.5)); \n' + 
+					'	} \n' + 
+					'	else { \n' + 
+					'		gl_FragColor = u_hasTexture ? (materialColor * texture2D(s_texture, v_texCoord)) : materialColor; \n' + 
+					'	} \n' + 
+					'}';
+	this.picking_vs =	'#ifdef GL_ES \n' + 
+						'	precision mediump float; \n' + 
+						'#endif	\n' + 
+						'\n' + 
+						'uniform mat4 u_transformMatrix; \n' + 
+						'attribute vec3 a_position; \n' + 
+						'\n' + 
+						'void main(void) { \n' + 
+						'	gl_Position = u_transformMatrix * vec4(a_position, 1.0); \n' + 
+						'}'; 
+	this.picking_fs =	'#ifdef GL_ES \n' + 
+						'	precision mediump float; \n' + 
+						'#endif	\n' + 
+						'\n' + 
+						'uniform vec3 u_pickingId; \n' + 
+						'\n' + 
+						'void main(void) { \n' + 
+						'	gl_FragColor = vec4(u_pickingId, 1.0); \n' + 
+						'}';
+
+	function createProgram(gl, vSrc, fSrc) {
+		var vShader = gl.createShader(gl.VERTEX_SHADER);
+		gl.shaderSource(vShader, vSrc);
+		gl.compileShader(vShader);
+		if(!gl.getShaderParameter(vShader, gl.COMPILE_STATUS)) {
+			if(JSC3D.console)
+				JSC3D.console.logWarning('Error occured in shader compilation: ' + gl.getShaderInfoLog(vShader) + ' The corresponding program will be ignored.');
+			return null;
+		}
+
+		var fShader = gl.createShader(gl.FRAGMENT_SHADER);
+		gl.shaderSource(fShader, fSrc);
+		gl.compileShader(fShader);
+		if(!gl.getShaderParameter(fShader, gl.COMPILE_STATUS)) {
+			if(JSC3D.console)
+				JSC3D.console.logWarning('Error occured in shader compilation: ' + gl.getShaderInfoLog(fShader) + ' The corresponding program will be ignored.');
+			return null;
+		}
+
+		var program = gl.createProgram();
+		gl.attachShader(program, vShader);
+		gl.attachShader(program, fShader);
+		gl.linkProgram(program);
+		if(!gl.getProgramParameter(program, gl.LINK_STATUS)) {
+			if(JSC3D.console)
+				JSC3D.console.logWarning('Error occured when generating program: ' + gl.getProgramInfoLog(program) + ' This program will be ignored.');
+			return null;
+		}
+
+		program.attributes = {};
+		var numOfAttribs = gl.getProgramParameter(program, gl.ACTIVE_ATTRIBUTES);
+		for(var i=0; i<numOfAttribs; i++) {
+			var attrib = gl.getActiveAttrib(program, i);
+			program.attributes[attrib.name] = gl.getAttribLocation(program, attrib.name);
+		}
+
+		program.uniforms = {};
+		var numOfUniforms = gl.getProgramParameter(program, gl.ACTIVE_UNIFORMS);
+		for(var i=0; i<numOfUniforms; i++) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Rename "i" as this name is already used in declaration at line 226. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS2814)